File: isilibrary/src/main/java/com/isi/isilibrary/dialog/RapidEditText.java
package com.isi.isilibrary.dialog;

import android.content.Context;
import android.text.InputType;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.textfield.TextInputEditText;

public class RapidEditText extends TextInputEditText {

    public RapidEditText(@NonNull Context context) {
        super(context);
    }

    public RapidEditText(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public RapidEditText(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setEditTextMail(){
        setInputType(InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS);
    }

    public void setEditTextNumber(boolean decimal, boolean signed){

        int type = InputType.TYPE_CLASS_NUMBER;

        if(decimal){
            type = type | InputType.TYPE_NUMBER_FLAG_DECIMAL;
        }

        if(signed){
            type = type | InputType.TYPE_NUMBER_FLAG_SIGNED;
        }

        setInputType(type);
    }

    public String getTextOrEmpty(){
        if(this.getText() == null){
            return "";
        }else{
         return this.getText().toString();
        }
    }
}

File: isilibrary/src/main/java/com/isi/isilibrary/dialog/RapidEditText.kt
package com.isi.isilibrary.dialog

import android.content.Context
import android.text.InputType
import android.util.AttributeSet
import com.google.android.material.textfield.TextInputEditText

class RapidEditText : TextInputEditText {
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    fun setEditTextMail() {
        inputType = InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
    }

    fun setEditTextNumber(decimal: Boolean, signed: Boolean) {
        var type = InputType.TYPE_CLASS_NUMBER
        if (decimal) {
            type = type or InputType.TYPE_NUMBER_FLAG_DECIMAL
        }
        if (signed) {
            type = type or InputType.TYPE_NUMBER_FLAG_SIGNED
        }
        inputType = type
    }

    val textOrEmpty: String
        get() = if (this.text == null) {
            ""
        } else {
            this.text.toString()
        }
}